src/modules/snowflake/data-connector/lambda_connectors/attribute_property_value_reader_by_entity.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from utils import udqw_constants
+from utils.connection_utils import connect_snowflake
+from utils.param_parser import UDQWParamsParser
+
+VALUE_TYPE_DATA_TYPE_MAPPING = {
+    'DOUBLE': 'doubleValue',
+    'LONG': 'longValue',
+    'INTEGER': 'intValue',
+    'STRING': 'stringValue',
+    'BOOLEAN': 'booleanValue'
+}
+
+# Configure logger
+LOGGER = logging.getLogger()
+LOGGER.setLevel(logging.INFO)
+
+# Connect to Snowflake
+SNOWFLAKE_CONNECTION = connect_snowflake()
+
+
+def lambda_handler(event, context):
+    """
+    Query attribute value for multiple properties from Snowflake.
+    Each property has single primitive value or list of primitive values.
+    The connector will return results with a map containing property name and its value.
+    """
+
+    # LOGGER.info('Receiving request: %s', event)
+
+    # 1. Parse input parameter
+    param_parser = UDQWParamsParser(event)
+
+    table_name = param_parser.get_attribute_property_table_name()
+    entity_id = param_parser.get_entity_id()
+    component_name = param_parser.get_component_name()
+    selected_properties = param_parser.get_selected_properties()
+    element_id = param_parser.get_element_id()
+    properties = param_parser.get_properties()
+
+    # 2. Get selected property definitions
+    property_definitions = {}
+    for property_name in selected_properties:
+        property_type = properties[property_name][udqw_constants.PROPERTY_DEFINITION][udqw_constants.PROPERTY_DATA_TYPE][udqw_constants.PROPERTY_TYPE]
+        property_definitions[property_name] = property_type
+
+    # 3. Generate query statement
+    wrapped_properties = ['identifier(\'' + property + '\')' for property in selected_properties]
+    query_statement = 'select {} from identifier(?) where elem_id = ?'.format(', '.join(wrapped_properties))
```

Comment:
is there sql-injection risk? should we use https://github.com/aws-samples/aws-iot-twinmaker-samples/blob/main/src/libs/udq_helper_utils/udq_utils/sql_detector.py similar to https://github.com/aws-samples/aws-iot-twinmaker-samples/blob/main/src/modules/timestream_telemetry/lambda_function/udq_data_reader.py#L69 ?

Reply:
Snowflake `identifier(?)` helps to bind variables https://docs.snowflake.com/en/sql-reference/identifier-literal.html